ludwig/data/dataset/ray.py

```diff
@@ -80,6 +86,14 @@ def __init__(
         self.data_hdf5_fp = training_set_metadata.get(DATA_TRAIN_HDF5_FP)
         self.data_parquet_fp = training_set_metadata.get(DATA_TRAIN_PARQUET_FP)
         self._processed_data_fp = df if isinstance(df, str) else None
+        self.window_size_bytes = None
```

Comment:
Instead of setting the window size bytes here, can you instead use a parameter for `auto_window`? If the parameter is set to `True`, then we can set `window_size_bytes` inside the `pipeline` function to use the heuristic below.